src/main/java/edu/kis/powp/jobs2d/features/DriverFeature.java

```diff
@@ -33,6 +34,7 @@ public static void setupDriverPlugin(Application application) {
 	public static void addDriver(String name, Job2dDriver driver) {
 		SelectDriverMenuOptionListener listener = new SelectDriverMenuOptionListener(driver, driverManager);
 		app.addComponentMenuElement(DriverFeature.class, name, listener);
+		driverManager.getPublisher().addSubscriber(new DriverLabelObserver());
```

Comment:
The observer should be subscribed only once. Now it is done each time a driver is added application.

Reply:
@kowallus I think the current solution should solve the problem